src/main/java/net/glowstone/GlowServer.java

```diff
@@ -1659,6 +1660,23 @@ public boolean suggestPlayerNamesWhenNullTabCompletions() {
         return false;
     }
 
+    @Override
+    public GlowPlayerProfile createProfile(UUID id) {
+        checkNotNull(id);
+        return createProfile(id, null);
+    }
+
+    @Override
+    public GlowPlayerProfile createProfile(String name) {
+        checkNotNull(name);
+        return createProfile(null, name);
```

Comment:
The null preconditions here were fine. I was talking about a problem with the uuid. This method creates a `GlowPlayerProfile` with `uuid=null` and `name=name`, which is currently not possible because of the [precondition](https://github.com/GlowstoneMC/Glowstone/pull/817/files#diff-bd59b1c2be78b091d2e887be7f414fe2R60) in the GlowPlayerProfile constructor. So the `createProfile` method with just a name would never work.

The `GlowPlayerProfile` is currently not intended to work with a null uuid. It would throw a null pointer in the `toNbt()` method and I'm not sure if some other classes can handle it if `getUniqueId()` returns null.